src/FGame.cpp

```diff
@@ -108,15 +105,20 @@ void FGame::render()
 		}
 	}
 			
-	// Render characters
+	// Render characters and their weapons
 	for (auto &character : m_engine->getCharacters())
 	{
-		auto sprite = loadSprite(m_characterTypeMap[character.getType()], character.getSize().x, character.getSize().y);
-		sprite->setPosition(sf::Vector2f(character.getPosition().x, character.getPosition().y));
-		m_window->draw(*sprite);
-		
-		// Render current weapon
-		//TODO
+		if (character.getHealth() > 0)
+		{
+			auto sprite = loadSprite(m_characterTypeMap[character.getType()], character.getSize().x, character.getSize().y);
+			sprite->setPosition(sf::Vector2f(character.getPosition().x, character.getPosition().y));
+			m_window->draw(*sprite);
+			
+			// auto weapon = character.getWeapon();
+			// auto weaponSprite = loadSprite(m_weaponTypeMap[weapon.getType()], weapon.getSize().x, weapon.getSize().y);
+			// weaponSprite->setPosition(sf::Vector2f(character.getPosition().x + character.getSize().x / 2.f, character.getPosition().y + character.getSize().y / 2.f));
+			// m_window->draw(*weaponSprite);
```

Comment:
This segfaults for some reason. If you have time, you could try to see what's wrong.
(I know now that `weapon.getSize()` is "wrong" because its size is unfortunately never set in the FWeapon constructor as it should, but that's not the only problem.)